drivers/hw_cc3xx/hw_cc3xx.c

```diff
@@ -23,8 +23,10 @@ static int hw_cc3xx_init_internal(void)
 	int res;
 
 	/* Initialize the cc3xx HW with or without RNG support */
-#if CONFIG_ENTROPY_CC3XX
+#if CONFIG_ENTROPY_CC3XX || defined(CONFIG_PSA_NEED_CC3XX_CTR_DRBG_DRIVER)
```

Comment:
This seems wrong and it seems that it was wrong before as well.

ENTROPY_CC3XX refers to the Zephyr driver which provided entropy from CC3XX. 

PSA_NEED_CC3XX_CTR_DRBG_DRIVER refers to the PSA CC3XX driver. 

To me the only actual requirement which seems logical to me here is: NRF_CC3XX_PLATFORM and we don't seem to have an option for not initializing the RNG.

I just want to make sure that we both understand all of this.  It's actually up to you how much refactoring you want to do. If you want to keep it like this that's fine as long as you add a comment to explain that you used these Kconfigs because of lack of others.